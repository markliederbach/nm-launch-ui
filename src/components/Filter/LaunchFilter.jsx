import React, { Component } from "react";
import { Form, Label, Input } from "semantic-ui-react";

export class LaunchFilter extends Component {
  constructor(props) {
    super(props);
    this.state = {
      filterValue: ""
    };
  }

  onChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };
  onKeyUp = () => {
    this.props.onFilter({ name: this.state.filterValue });
  };

  render() {
    return (
      <Form>
        <Input
          focus
          placeholder="Filter by keyword..."
          name="filterValue"
          type="text"
          onChange={this.onChange}
          onKeyUp={this.onKeyUp}
        />
      </Form>
    );
  }
}
