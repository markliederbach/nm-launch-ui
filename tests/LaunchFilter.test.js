import React from "react";
import { shallow, configure } from "enzyme";
import renderer from "react-test-renderer";
import { LaunchFilter } from "../src/components/Filter/LaunchFilter";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("LaunchFilter", function() {
  let spy, wrapper;
  beforeAll(() => {
    spy = jest.fn();
  });
  beforeEach(() => {
    spy.mockReset();
    wrapper = shallow(<LaunchFilter onFilter={spy} />);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<LaunchFilter />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("calls onFilter after keystroke", () => {
    wrapper
      .find("Input")
      .first()
      .simulate("change", {
        target: { name: "filterValue", value: "Falcon" }
      });
    wrapper
      .find("Input")
      .first()
      .simulate("keyup");
    expect(spy).toBeCalledWith({ name: "Falcon" });
    expect(wrapper.state("filterValue")).toEqual("Falcon");
  });
});
