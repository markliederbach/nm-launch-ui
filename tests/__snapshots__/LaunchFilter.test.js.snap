// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchFilter renders correctly 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="ui focus input"
  >
    <input
      name="filterValue"
      onChange={[Function]}
      onKeyUp={[Function]}
      placeholder="Filter by name..."
      type="text"
    />
  </div>
</form>
`;
